### PR TITLE
feat: remove empty qualified and at rules

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Feat: remove empty qualified rules and at-rules

I think we should refactor `AtRule` AST and use only one AtRule structure with different block and prelude types, it will reduces logic for minified and codegen and do our AST is simple, I am working on it

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_minifier/src/compress/empty.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            | Rule::AtRule(AtRule::Supports(SupportsRule { block, .. }))
+            | Rule::AtRule(AtRule::Page(PageRule { block, .. }))
+            | Rule::AtRule(AtRule::Viewport(ViewportRule { block, .. }))
+            | Rule::AtRule(AtRule::Document(DocumentRule { block, .. }))
+            | Rule::AtRule(AtRule::Property(PropertyRule { block, .. }))
+                if block.value.is_empty() =>
+            {
+                return false;
+            }
+            Rule::AtRule(AtRule::Layer(LayerRule { block, .. })) if block.is_some() => {
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compress/empty.rs:28]: I think it can be `Rule::AtRule(AtRule::Layer(LayerRule { block: Some(block), .. }))` instead.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Mostly LGTM. Thank you!

**Comment by alexander-akait** [comment]
@kdy1 Done and rebased

**Review by kdy1 — APPROVED** [review]
LGTM.

---


swc-bump:
 - swc_css_minifier

**Pull request merged**